server/hotrod/src/main/java/org/infinispan/server/hotrod/LocalContextHandler.java

```diff
@@ -25,12 +27,32 @@ public LocalContextHandler(NettyTransport transport) {
    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
       if (msg instanceof CacheDecodeContext) {
          CacheDecodeContext cdc = (CacheDecodeContext) msg;
-         realChannelRead(ctx, msg, cdc);
+         if (cdc.header.op.isMultimap())
```

Comment:
I have no idea why this code is here... and I have no idea why this class exists... /cc @wburns

Reply:
If I remember correctly this class is here to make operations that should not require blocking to be faster as they don't have to use a separate thread (in our case GET should always be local as we have smart routing for example). Let me confirm that though.

Reply:
Yeah that is what it is for https://github.com/karesti/infinispan/commit/a7526f4904f0f9e75f8e241f650f26820f1f41da

So in this case what she has for the Get should be fine.

So basically you want to add operations that you know won't block to this handler. And if it can block it should go in the standard [ContextHandler](https://github.com/infinispan/infinispan/blob/master/server/hotrod/src/main/java/org/infinispan/server/hotrod/ContextHandler.java#L61)

Reply:
@wburns I might need to change to use ReadEntryView instead of ReadWriteEntryView or we don't care about that ?

Reply:
@wburns the this is that I'm using the ReadWriteMap functional interface, even when I call just "get" (I don't change any value, but I'm calling that). Can this harm or the is no link at all ?

Reply:
Not sure I understand. As long as the operation isn't going to block this thread we can leave it here. The object is all stored in 1 node and it should be this node so any read should be fine.  It shouldn't matter that it is a write, the problem is that a write normally has to go remote (for example a LOCAL only cache in hotrod could use this handler as well)